cblox/include/cblox/core/tsdf_esdf_submap.h

```diff
@@ -0,0 +1,55 @@
+//
+// Created by victor on 07.01.19.
+//
+
+#ifndef CBLOX_CORE_TSDF_ESDF_SUBMAP_H_
+#define CBLOX_CORE_TSDF_ESDF_SUBMAP_H_
+
+#include <memory>
+#include "cblox/core/tsdf_submap.h"
+
+namespace cblox {
+
+class TsdfEsdfSubmap : public TsdfSubmap {
+ public:
+  typedef std::shared_ptr<TsdfEsdfSubmap> Ptr;
+  typedef std::shared_ptr<const TsdfEsdfSubmap> ConstPtr;
+
+  struct Config : TsdfSubmap::Config, EsdfMap::Config {};
+
+  TsdfEsdfSubmap(const Transformation &T_M_S, SubmapID submap_id, Config config)
+      : TsdfSubmap(T_M_S, submap_id, config), config_(config) {
+    esdf_map_.reset(new EsdfMap(config));
+  }
+
+  ~TsdfEsdfSubmap() {
+    if (!esdf_map_.unique()) {
+      std::cout << "Underlying esdf map from SubmapID: " << submap_id_
```

Comment:
Hmmm. at some point I was worried about leaks apparently... Lets leave these in for now... Schedule for removal. Maybe add a comment to that effect.